plugins/woocommerce/src/Admin/API/Reports/Customers/Controller.php

```diff
@@ -523,6 +524,22 @@ public function get_collection_params() {
 				'type' => 'integer',
 			),
 		);
+		$params['filter_empty']            = array(
+			'description'       => __( 'Filter out results where all passed fields are empty', 'woocommerce' ),
+			'type'              => 'array',
+			'validate_callback' => 'rest_validate_request_arg',
+			'items'             => array(
+				'type' => 'string',
+				'enum' => array(
+					'email',
+					'name',
+					'country',
+					'city',
+					'state',
+					'postcode',
```

Comment:
Might be worth adding the `username` field here too.

Reply:
Can be username empty? If not, then probably this won't bring much

Reply:
Yeah, any customer that checks out as a guest will not have a username.